### PR TITLE
Fix #45594: Segment name is missing from the segment revision history title

### LINKED ISSUE #45594: Datamodel segment history title references a table name instead of the actual segment name
### Describe the bug

Datamodel segment history title references a table name instead of the actual segment name.
It has been like that since forever.

### To Reproduce

Have a segment based on Orders table, and give it a name (in my case it was "Orders < 100"

1. Go to `/admin/datamodel/segment/:id/revisions` 
2. Notice that the title references the table (Orders) instead of a segment
image

### Expected behavior

The title should reference a segment name.

### Logs

_No response_

### Information about your Metabase installation

```JSON
local dev, `master`, 9c02ca6, H2, Sample Database
```

### Severity

P3

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### What does this PR accomplish?
- Fixes #45594 
- Reproduces that issue to prevent future regressions
- Removes redundant `table` prop

### Stress-test
50x https://github.com/metabase/metabase/actions/runs/9948179365

### CODE REVIEW

File: frontend/src/metabase/admin/datamodel/containers/RevisionHistoryApp.jsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default connect(mapStateToProps, mapDispatchToProps)(RevisionHistoryApp);
 class SegmentRevisionHistoryInner extends Component {
   render() {
     const { segment, ...props } = this.props;
-    return <RevisionHistory object={segment} {...props} />;
+    return <RevisionHistory segment={segment} {...props} />;
```
Comment (nemanjaglumac) [inline frontend/src/metabase/admin/datamodel/containers/RevisionHistoryApp.jsx:35]: The actual fix.

File: frontend/src/metabase/admin/datamodel/components/revisions/RevisionDiff.jsx [hunk 2]
```diff
@@ -42,7 +40,7 @@ export default class RevisionDiff extends Component {
           </div>
           <div>
             {this.props.property === "definition" ? (
-              <QueryDiff diff={this.props.diff} tableMetadata={tableMetadata} />
+              <QueryDiff diff={this.props.diff} />
```
Comment (nemanjaglumac) [inline frontend/src/metabase/admin/datamodel/components/revisions/RevisionDiff.jsx:43]: You can verify this by going to the `QueryDiff` - no `tableMetadata` is even mentioned there.

File: e2e/test/scenarios/admin/datamodel/datamodel.cy.spec.js [hunk 3]
```diff
@@ -880,41 +880,56 @@ describe("scenarios > admin > datamodel > segments", () => {
...
 
-    it("should show segment revision history (metabase#45577)", () => {
+    it("should show segment revision history (metabase#45577, metabase#45594)", () => {
       cy.request("PUT", "/api/segment/1", {
         description: "Medium orders",
         revision_message: "Foo",
       });
 
+      cy.log("Make sure revisions are displayed properly in /references");
+      cy.visit("/reference/segments/1/revisions");
```
Comment (kamilmielnik) [inline e2e/test/scenarios/admin/datamodel/datamodel.cy.spec.js:890]: Please avoid multiple `cy.visit` calls in a single test unless it's needed for repro - it's the most expensive action.
The test will almost always run faster when _clicking_ through UI instead.

File: frontend/src/metabase/admin/datamodel/containers/RevisionHistoryApp.jsx (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase/admin/datamodel/containers/RevisionHistoryApp.jsx:35]: The actual fix was to remove `Tables.load` which injected `object` prop into `RevisionHistory` overriding whatever was being passed here.

**Review by kamilmielnik — APPROVED** [review]

File: frontend/src/metabase/admin/datamodel/containers/RevisionHistoryApp.jsx (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/admin/datamodel/containers/RevisionHistoryApp.jsx:35]: The fix was working even before I removed Tables simply by changing the name of the prop from `object` to `segment`. Tables were overriding the `object` prop.

File: e2e/test/scenarios/admin/datamodel/datamodel.cy.spec.js (hunk 3 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/admin/datamodel/datamodel.cy.spec.js:890]: Got it, but in this case this was the most effective path.

**Pull request merged**